From 10041953dd0ad4938836496e72219f335c6d8ca4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Sun, 14 Sep 2025 04:08:44 +0200
Subject: [PATCH] Prepare multiprocessing handling for Python 3.14

Python 3.14 changed multiprocessing method from 'fork' to 'forkserver'
on Linux too:
https://github.com/python/cpython/issues/84559

This leads to issue similar to when Python 3.8 changed it on Mac OS:
https://github.com/saltstack/salt/issues/57742

Change the condition to check for != 'fork' instead of == 'spawn'.
---
 changelog/68148.changed.md | 1 +
 salt/utils/platform.py     | 8 ++++++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/salt/utils/platform.py b/salt/utils/platform.py
index 59a04b451bcd..1ed145ab1e3d 100644
--- a/salt/utils/platform.py
+++ b/salt/utils/platform.py
@@ -237,12 +237,16 @@ def is_aarch64():
 
 def spawning_platform():
     """
-    Returns True if multiprocessing.get_start_method(allow_none=False) returns "spawn"
+    Returns True if multiprocessing.get_start_method(allow_none=False) != "fork"
+    ("spawn" or "forkserver")
 
     This is the default for Windows Python >= 3.4 and macOS on Python >= 3.8.
     Salt, however, will force macOS to spawning by default on all python versions
+    Starting with Python 3.14, Linux also defaults to not "fork", but it uses
+    (new) "forkserver" instead of "spawn". Functionally, it's very similar as
+    far as Salt is concerned (process state is not inherited).
     """
-    return multiprocessing.get_start_method(allow_none=False) == "spawn"
+    return multiprocessing.get_start_method(allow_none=False) != "fork"
 
 
 def get_machine_identifier():
